chore: add database testing guide (#9921)
supabase/supabase: apps/reference/docs/guides/cli/testing.mdx

---
id: testing

> title: Testing Database

Reviewer: ```suggestion
title: Testing your Database
```

> description: How to test your schema changes locally in the database.

Reviewer: ```suggestion
description: Test your database schema, tables, functions, and policies.
```

---

import Tabs from '@theme/Tabs'
import TabItem from '@theme/TabItem'

> You can use the Supabase CLI to test your database. Make sure you have [installed the CLI](https://github.com/supabase/cli#getting-started) on your local machine and [linked](https://supabase.com/docs/reference/cli/usage#supabase-link) your Supabase project.

Reviewer: ```suggestion
You can use the Supabase CLI to test your database. To get started:

- [install the CLI](https://github.com/supabase/cli#getting-started) on your local machine 
- [link](/docs/reference/cli/usage#supabase-link) your Supabase project
```

Reviewer: @sweatybridge perhaps we can move the installation instructions to the docs? (doesn't need to be in this PR, just in the future)

Author: Sure, I will link to overview which has installation instructions. https://supabase.com/docs/guides/cli

## Creating a test

Create a tests folder inside the supabase folder:

```bash
mkdir -p ./supabase/tests/database
```

> Create a new file with a test with the `.sql` extension.

Reviewer: ```suggestion
Create a new file with the `.sql` extension which will contain the test.
```
